rpc/src/client/event_listener.rs

```diff
@@ -83,33 +88,81 @@ impl EventListener {
         Ok(())
     }
 
-    /// Get the next event from the websocket
-    pub async fn get_event(&mut self) -> Result<Option<ResultEvent>, RPCError> {
-        let msg = self
-            .socket
-            .next()
-            .await
-            .ok_or_else(|| RPCError::websocket_error("web socket closed"))??;
+    /// Get the next event from the websocket. Automatically handles websocket
+    /// protocol details, like responding to ping messages, so it can either
+    /// produce events or errors.
+    pub async fn get_event(&mut self) -> Result<ResultEvent, RPCError> {
```

Comment:
I think it would be much nicer if the `subscribe` would actually return events (a stream of events) for that particular subscription. Then, the even-listener would also run a "infinite" loop where it would capture control messages but it would route the only the relevant events to the particular subscriptions. 
But that could probably be done in a separate PR.

Reply:
Yes, that sounds like a much more intuitive developer interface. I'll make a note of that to work on that in a future PR 👍

Reply:
Just came to me now: this sounds like what you've recommended on #313, right?

Reply:
The issue with this design is, that it hinges on the caller to continuously drive the websocket event machinery. Likely expected to be run in a loop. Which can lead to subtle bugs, when we imagine the caller not calling `get_event` in time and the required `Pong` message to keep the connection alive is never fired. For a user to understand why the connection was closed will be hard.

IMHO we need to rework this so the event loop is running continuously and acts more like a stream. We can still build a synchronous API on top of it, if that is desired.